e2e-tests/docs/CI.md

```diff
@@ -0,0 +1,135 @@
+
+## GitHub Pull Requests
+
+When a new Pull Request (PR) is opened at [backstage-showcase](https://github.com/janus-idp/backstage-showcase), tests can be triggered in two ways:
+
+1. **Commenting `/ok-to-test`**: Only members of the janus-idp GitHub organization can mark the PR with this comment to validate it for testing.
+
+2. **Triggering Tests Post-Validation**: After a janus-idp member has validated the PR, anyone can trigger tests using the following commands:
+   - `/test` or `/test all`
+   - `/retest`
+
+These interactions are picked up by the OpenShift-CI service, which sets up a test environment on the **IBM Cloud**, specifically on the `rhdh-pr-os` OpenShift Container Platform (OCP) cluster. The configurations and steps for setting up this environment are defined in the `openshift-ci-tests.sh` script.
+
+> **Note:** We do **not** have PR checks running on Azure Kubernetes Service (AKS); all PR checks are executed on IBM Cloud's `rhdh-pr-os` cluster.
+
+Detailed steps on how the tests and reports are managed can be found in the `run_tests()` function within the `openshift-ci-tests.sh` script. Additionally, all the different `yarn` commands that trigger various Playwright projects are described in the `package.json` file.
+
+When the test run is complete, the status will be reported under your PR checks.
+
+> **Important:** The environment in which the PR tests are executed is shared and ephemeral. All PR tests queue for the same environment, which is destroyed and recreated for each PR. Test outputs (screenshots, recordings, walkthroughs, etc.) are stored for a retention period of **6 months** and can be accessed by checking the **Details -> Artifacts** of the test check on the PR.
+
+### Retrying Tests
+
+PR tests are not automatically retried beyond the individual test retries specified in the Playwright configuration (each test is retried up to 2 times on failure). However, you can manually retrigger the entire test suite by commenting:
+
+- `/retest e2e-tests`
+- `/test all`
+
+This is useful if you believe a failure was due to a flake or external issue and want to rerun the tests without making any code changes.
+
+### CI Job Definitions
+
+#### Pull Request Test Job
+
+- **Purpose:** Validate new PRs for code quality, functionality, and integration.
+- **Trigger:** When a PR is opened and `/ok-to-test` is commented by a janus-idp member, or when `/test`, `/test all`, or `/retest` is issued after validation.
+- **Environment:** Runs on the ephemeral `rhdh-pr-os` cluster on IBM Cloud.
+- **Configurations:**
+  - Tests are executed on both **RBAC** (Role-Based Access Control) and **non-RBAC** namespaces.
+- **Steps:**
+  1. **Detection:** OpenShift-CI detects the PR event.
+  2. **Environment Setup:** The test environment is set up using the `openshift-ci-tests.sh` script.
```

Comment:
Maybe it is outside the scope, but there is also the test runner (based on this image: https://github.com/janus-idp/backstage-showcase/blob/main/.ibm/images/Dockerfile) that is being set up. 
The definitions for OpenShift CI, for PR checks and nightly, are here:
- https://github.com/openshift/release/tree/master/ci-operator/jobs/janus-idp/backstage-showcase
- https://github.com/openshift/release/tree/master/ci-operator/config/janus-idp/backstage-showcase
- https://github.com/openshift/release/tree/master/ci-operator/step-registry/janus-idp/backstage-showcase
You can also see how the `openshift-ci-tests.sh` gets executed

Reply:
really interesting stuff!! Will check it out and add it for sure!